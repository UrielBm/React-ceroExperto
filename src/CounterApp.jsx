import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
 
const CounterApp=({value}) => {
    const [counter,setCounter]= useState(value);
    //Funcion del boton 
    const handleAdd = () =>{
        setCounter(counter+1);
        // setCounter((c)=>c+1);
    }
    const Reset= () =>{
        setCounter(value);
    }
    const Handleless=() =>{
        setCounter(counter-1);
    }
    return (
        <Fragment>
        <h1>CouterApp *tarea*</h1>
        {/* <pre>{JSON.stringify(saludo,null, 3 )}</pre> */}
        <h2>{counter}</h2>
        <div className="wrapper-botones">
        <button className="button" onClick={handleAdd}>+1</button>
        <button className="button" onClick={Reset}>Reset</button>
        <button className="button" onClick={Handleless}>-1</button>
        </div>
        </Fragment>
    ); 
}
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
CounterApp.defaultProps={
    value: 10
}
export default CounterApp;