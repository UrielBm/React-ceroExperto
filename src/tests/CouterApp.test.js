import React from 'react';
import '@testing-library/jest-dom';
// import {render} from '@testing-library/react';
import { shallow } from 'enzyme';
import CouterApp from '../CounterApp';
import CounterApp from '../CounterApp';
describe('Probando la correcta funcionalidad de CouterApp.js', () => {
         let wrapperCouterApp=shallow(<CouterApp />);
        beforeEach(() =>{
            wrapperCouterApp=shallow(<CouterApp />);
        });
    test('Debe mostar correctamente el componente <CouterApp /> con su valor por defecto 10 ', () => {
        expect(wrapperCouterApp).toMatchSnapshot();
    });
    test('Debe mostar correctamente el componente <CouterApp /> enviando el valor mediante props  ', () => {
        const v=100;
        const wrapperCouterApp=shallow(<CounterApp value={v} />);
        const ContadorH2=wrapperCouterApp.find('h2').text().trim();
        // console.log(ContadorH2);
        expect(ContadorH2).toBe('100');

    });
    test('Debe incrementar +1 con el boton ', () => {
       wrapperCouterApp.find('button').at(0).simulate('click');
        // console.log(btn1.html()) para ocupar agegar const btn1=
        const ContadorH2=wrapperCouterApp.find('h2').text().trim();
        expect(ContadorH2).toBe('11');
    });
    test('Debe decrementar -1 con el boton ', () => {
        const btn3=wrapperCouterApp.find('button').at(2).simulate('click');
         // console.log(btn1.html())
         const ContadorH2=wrapperCouterApp.find('h2').text().trim();
         expect(ContadorH2).toBe('9');
     });
     test('Retorna los valores que se le mandan en el props  con el boton reset', () => {
        const v=110;
        const wrapperCouterApp=shallow(<CounterApp value={v} />);
        wrapperCouterApp.find('button').at(0).simulate('click');
        wrapperCouterApp.find('button').at(0).simulate('click');
        wrapperCouterApp.find('button').at(1).simulate('click');
        const ContadorH2=wrapperCouterApp.find('h2').text().trim();
        console.log(ContadorH2);
        expect(ContadorH2).toBe('110');
     });
     
    
})
