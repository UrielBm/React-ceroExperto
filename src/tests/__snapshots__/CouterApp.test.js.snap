// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando la correcta funcionalidad de CouterApp.js Debe mostar correctamente el componente <CouterApp /> con su valor por defecto 10  1`] = `
<Fragment>
  <h1>
    CouterApp *tarea*
  </h1>
  <h2>
    10
  </h2>
  <div
    className="wrapper-botones"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      +1
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      className="button"
      onClick={[Function]}
    >
      -1
    </button>
  </div>
</Fragment>
`;
